code/modules/reagents/chemistry/reagents/alcohol_reagents.dm

```diff
@@ -1892,6 +1892,9 @@ All effects don't start immediately, but rather get worse over time; the rate is
 /datum/reagent/consumable/ethanol/fanciulli/on_mob_metabolize(mob/living/M)
 	if(M.health > 0)
 		M.adjustStaminaLoss(20)
+		if(iscarbon(M))
+			var/mob/living/carbon/C = M
+			C.clear_stamina_regen()
```

Comment:
``clear_stamina_regen`` should probably be on the living level with ``adjustStaminaLoss``, they are both for stamina so it makes no sense for them to be on different levels. iscarbon check every time looks terrible.